components/channel_notifications_modal/channel_notifications_modal.tsx

```diff
@@ -35,279 +64,301 @@ type Props = PropsFromRedux & {
-            this.updateSection(NotificationSections.NONE);
-        }
+    function handleExit() {
+        props.onExited();
     }
 
-    handleSubmitDesktopNotifyLevel = () => {
-        const channelNotifyProps = this.props.channelMember && this.props.channelMember.notify_props as ChannelMemberNotifyProps;
-        const {desktopNotifyLevel, desktopThreadsNotifyLevel} = this.state;
-
-        if (
-            channelNotifyProps?.desktop === desktopNotifyLevel &&
-            channelNotifyProps?.desktop_threads === desktopThreadsNotifyLevel
-        ) {
-            this.updateSection(NotificationSections.NONE);
-            return;
+    const resetToDefault = useCallback((settingName: string) => {
+        const defaultSettings = props.currentUser.notify_props;
+        if (settingName === 'desktop') {
+            setSettings({...settings, desktop: defaultSettings.desktop, desktop_threads: defaultSettings.desktop_threads || settings.desktop_threads});
         }
-
-        const props = {desktop: desktopNotifyLevel, desktop_threads: desktopThreadsNotifyLevel};
-
-        this.handleUpdateChannelNotifyProps(props);
-    }
-
-    handleUpdateDesktopNotifyLevel = (desktopNotifyLevel: ChannelNotifyProps['desktop']) => this.setState({desktopNotifyLevel});
-
-    handleUpdateDesktopThreadsNotifyLevel = (desktopThreadsNotifyLevel: UserNotifyProps['desktop_threads']) => this.setState({desktopThreadsNotifyLevel});
-
-    handleSubmitMarkUnreadLevel = () => {
-        const channelNotifyProps = this.props.channelMember && this.props.channelMember.notify_props;
-        const {markUnreadNotifyLevel} = this.state;
-
-        if (channelNotifyProps?.mark_unread === markUnreadNotifyLevel) {
-            this.updateSection(NotificationSections.NONE);
-            return;
+        if (settingName === 'push') {
+            setSettings({...settings, push: defaultSettings.desktop, push_threads: defaultSettings.push_threads || settings.push_threads});
         }
-
-        const props = {mark_unread: markUnreadNotifyLevel};
-        this.handleUpdateChannelNotifyProps(props);
-    }
-
-    handleUpdateMarkUnreadLevel = (markUnreadNotifyLevel: ChannelNotifyProps['mark_unread']) => this.setState({markUnreadNotifyLevel});
-
-    handleSubmitPushNotificationLevel = () => {
-        const channelNotifyProps = this.props.channelMember && this.props.channelMember.notify_props as ChannelMemberNotifyProps;
-        const {pushNotifyLevel, pushThreadsNotifyLevel} = this.state;
-
-        if (
-            channelNotifyProps?.push === pushNotifyLevel &&
-            channelNotifyProps?.push_threads === pushThreadsNotifyLevel
-        ) {
-            this.updateSection(NotificationSections.NONE);
-            return;
+    }, [settings, props.currentUser]);
+
+    const handleChange = useCallback((values: Record<string, string>) => {
+        setSettings({...settings, ...values});
+    }, [settings]);
+
+    const handleMobileSettingsChange = useCallback(() => {
```

Comment:
Don't think we need to useCallback in this case as this child component doesn't continuously rerender. Reference here: https://www.knowledgehut.com/blog/web-development/all-about-react-usecallback